internal/pkg/status/legacy_status.go

```diff
@@ -0,0 +1,238 @@
+	clientu "sigs.k8s.io/cli-experimental/internal/pkg/client/unstructured"
+)
+
+// IsReadyFn - status getter
+type IsReadyFn func(*unstructured.Unstructured) (bool, error)
+
+var legacyTypes = map[string]map[string]IsReadyFn{
+	"": map[string]IsReadyFn{
+		"Service":               alwaysReady,
+		"Pod":                   podReady,
+		"PersistentVolumeClaim": pvcReady,
+	},
+	"apps": map[string]IsReadyFn{
+		"StatefulSet": stsReady,
+		"DaemonSet":   daemonsetReady,
+		"Deployment":  deploymentReady,
+		"ReplicaSet":  replicasetReady,
+	},
+	"policy": map[string]IsReadyFn{
+		"PodDisruptionBudget": pdbReady,
+	},
+	"batch": map[string]IsReadyFn{
+		"CronJob": cronjobReady,
+		"Job":     jobReady,
+	},
+}
+
+// GetLegacyReadyFn - True if we handle it as a known type
+func GetLegacyReadyFn(u *unstructured.Unstructured) IsReadyFn {
+	gvk := u.GroupVersionKind()
+	g := gvk.Group
+	k := gvk.Kind
+	if _, ok := legacyTypes[g]; ok {
+		if fn, ok := legacyTypes[g][k]; ok {
+			return fn
+		}
+	}
+	return nil
+}
+
+func alwaysReady(u *unstructured.Unstructured) (bool, error) { return true, nil }
+
+func compareIntFields(u *unstructured.Unstructured, field1, field2 []string, checkFuncs ...func(int, int) bool) (bool, error) {
+	v1, ok, err := clientu.NestedInt(u.UnstructuredContent(), field1...)
+	if err != nil {
+		return true, err
+	}
+	if !ok {
+		return false, fmt.Errorf("%v not found", field1)
+	}
+
+	v2, ok, err := clientu.NestedInt(u.UnstructuredContent(), field2...)
+	if err != nil {
+		return true, err
+	}
+	if !ok {
+		return false, fmt.Errorf("%v not found", field2)
+	}
+
+	rv := true
+
+	for _, fn := range checkFuncs {
+		rv = rv && fn(v1, v2)
+	}
+
+	return rv, nil
+}
+
+func equalInt(v1, v2 int) bool { return v1 == v2 }
+func geInt(v1, v2 int) bool    { return v1 >= v2 }
+
+// Statefulset
+func stsReady(u *unstructured.Unstructured) (bool, error) {
+	c1, err := compareIntFields(u, []string{"status", "readyReplicas"}, []string{"spec", "replicas"}, equalInt)
+	if err != nil {
+		return c1, err
+	}
+	c2, err := compareIntFields(u, []string{"status", "currentReplicas"}, []string{"spec", "replicas"}, equalInt)
+	if err != nil {
+		return c2, err
+	}
+	return c1 && c2, nil
+}
+
+// Deployment
+func deploymentReady(u *unstructured.Unstructured) (bool, error) {
+	progress := true
+	available := true
+	conditions := clientu.GetConditions(u.UnstructuredContent())
+
+	if len(conditions) == 0 {
+		return false, fmt.Errorf("no conditions in object")
+	}
+
+	for _, c := range conditions {
+		switch clientu.GetStringField(c, "type", "") {
+		case "Progressing": //appsv1.DeploymentProgressing:
+			// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/deployment/progress.go#L52
+			status := clientu.GetStringField(c, "status", "")
+			reason := clientu.GetStringField(c, "reason", "")
+			if status != "True" || reason != "NewReplicaSetAvailable" {
+				progress = false
```

Comment:
Set `progress` and `availalbe` to false and then only set them to true when this condition is met (i.e. progressing && NewReplicaSetAvailable). Otherwise if these condition types aren't found, `progress` and `availalbe` will still be set to true.

Reply:
refactored.